renderapi/tilespec.py

```diff
@@ -118,6 +119,55 @@ def bbox(self):
                 'undefined bounding box for tile {}'.format(self.tileId))
         return box
 
+    def bbox_transformed(self,ndiv_inner=0,tf_limit=None):
+        """method to return Nx2 transformed coordinates of bounding box
+        Paramters
+        ---------
+        ndiv_inner : starting with just corner points, add intermediate
+            points to the boundary, recursively, ndiv_inner times
+        tf_limit :
+            0 returns the raw bounding box
+            1 returns the bounding box with the first transform applied
+            ...
+            None all transforms are applied
+
+        Returns
+        -------
+        Nx2 array ready for input to shapely.Polygon()
+        """
+        #start with closed Nx2 array of corners
+        xy = np.zeros((5,2)).astype('float')
+        xy[0,:] = [0,0]
+        xy[1,:] = [0,self.height]
+        xy[2,:] = [self.width,self.height]
+        xy[3,:] = [self.width,0]
+        xy[4,:] = [0,0]
+    
+        #recursively add points to the boundary
+        while ndiv_inner>0:
+            sz = 2*xy.shape[0]-1
+            newxy = np.zeros((sz,2)).astype('float')
+            newxy[0::2,:] = xy[:,:]
+            newxy[1:sz:2,:] = 0.5*(newxy[0:(sz-2):2,:] + newxy[2:sz:2,:])
+            xy = newxy
+            ndiv_inner-=1
+
+        #set which transforms
+        if (tf_limit==None)|(tf_limit>len(self.tforms)):
+            tlist = self.tforms
+        else:
+            tlist = self.tforms[0:tf_limit]
+
+        #cycle through the tspec tforms and apply the transforms
+        for transform in tlist:
+            try:
+                xy = transform.tform(xy)
+            except Exception as e:
+                logger.error(e)
+                raise RenderError("transform class does not have tform method?")
```

Comment:
Probably better to specifically catch the AttributeError that would come from not having a tform method or the TypeError that could come from not having it be callable.  e.g.

```python
try:
    xy = transform.tform(xy)
except (AttributeError,TypeError) as e:
    raise RenderError('unacceptable transform instance: {}'.format(e))
```